presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -189,6 +192,8 @@
 
     private final ResolvedFunctionDecoder functionDecoder;
 
+    private static final int TABLE_REDIRECTION_MAX_TIMES = 100;
```

Comment:
@electrum I re-included the max redirection enforcement to detect programming bugs that may lead to query execution hang. I thought that such bugs would be more common for table redirection than catalog redirection. Please let me know what you think.

Reply:
Let's name this `MAX_TABLE_REDIRECTIONS`